### PR TITLE
Test/coverage

### PR DESCRIPTION
**Pull request by fpietka**
I added Istanbul to track code coverage and added tests for `url-formatter`.

For Istanbul, I changed the `test` task, but you can also use a separate task. Here is what it does:
- Output `text` and `text-summary`
- Writes coverage results using `lcov` format

It should be a great addition to plug to a CI of your choice to track code coverage.

The more coverage seems to be achieved testing socket related code. I know it can be done, but I must say I didn't succeed yet :grin: 

I also failed at testing `etag-handler` as it uses callback, but it may be a simple one to test.

### CODE REVIEW

File: tests/lib/github/url-formatter.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    it('should build full url with per_page option.', function () {
+      expect(formatter('/user/orgs', {per_page: 100})).to.eq('https://api.github.com/user/orgs?per_page=100');
+    });
+
+    it('should build full url ignoring unwanted options.', function () {
+      expect(formatter('/user/orgs', {foobar: 'baz'})).to.eq('https://api.github.com/user/orgs');
+    });
+
+    it('should return full url when it already egins with "http".', function () {
```
Comment (ludovic-gonthier) [inline tests/lib/github/url-formatter.js:19]: Little typo: egins => begins, I'll correct it when integrating the Coverage CI

**Comment by ludovic-gonthier** [comment]
:+1: I'll do the integration to [Coveralls](https://coveralls.io) when this get merged.

**Comment by ludovic-gonthier** [comment]
I've found the way to tests callback: https://github.com/seppevs/seppevs.github.io/blob/master/code-samples/unit_testing_with_mocks_in_node_js/twitstat/tests/twitstat.js#L55.

I thought it could have interested you

**Comment by fpietka** [comment]
I'll take a look, thanks!

**Pull request merged**